src/main/java/teammates/ui/controller/StudentCourseJoinAction.java

```diff
@@ -34,8 +35,13 @@ public ActionResult execute() throws EntityDoesNotExistException {
         
         String confirmUrl = Const.ActionURIs.STUDENT_COURSE_JOIN_AUTHENTICATED 
                 + "?" + Const.ParamsNames.REGKEY + "=" + regkey 
-                + "&" + Const.ParamsNames.NEXT_URL + "=" + nextUrl;
-        data = new StudentCourseJoinConfirmationPageData(account, student, confirmUrl, Logic.getLogoutUrl(confirmUrl));
+                + "&" + Const.ParamsNames.NEXT_URL + "=" + Sanitizer.sanitizeForNextUrl(nextUrl);
+        String nextUrlType = nextUrl.replaceFirst("^(/page/[A-Za-z]+)\\?.*$", "$1");
```

Comment:
Not that it can't be done, but... I don't really see the point why. The method is hardly that complicated to warrant its own test. Also I don't know where I should extract the method to. Doesn't seem to fit to any of the util class.